core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultNettyOptions.java

```diff
@@ -131,7 +134,30 @@ public ByteBufAllocator allocator() {
 
   @Override
   public void afterBootstrapInitialized(Bootstrap bootstrap) {
-    // nothing to do
+    boolean tcpNoDelay = config.getBoolean(DefaultDriverOption.SOCKET_TCP_NODELAY);
+    bootstrap.option(ChannelOption.TCP_NODELAY, tcpNoDelay);
+    if (config.isDefined(DefaultDriverOption.SOCKET_KEEP_ALIVE)) {
+      boolean keepAlive = config.getBoolean(DefaultDriverOption.SOCKET_KEEP_ALIVE);
+      bootstrap.option(ChannelOption.SO_KEEPALIVE, keepAlive);
+    }
+    if (config.isDefined(DefaultDriverOption.SOCKET_REUSE_ADDRESS)) {
+      boolean reuseAddress = config.getBoolean(DefaultDriverOption.SOCKET_REUSE_ADDRESS);
+      bootstrap.option(ChannelOption.SO_REUSEADDR, reuseAddress);
+    }
+    if (config.isDefined(DefaultDriverOption.SOCKET_LINGER_INTERVAL)) {
+      int lingerInterval = config.getInt(DefaultDriverOption.SOCKET_LINGER_INTERVAL);
+      bootstrap.option(ChannelOption.SO_LINGER, lingerInterval);
+    }
+    if (config.isDefined(DefaultDriverOption.SOCKET_RECEIVE_BUFFER_SIZE)) {
+      int receiveBufferSize = config.getInt(DefaultDriverOption.SOCKET_RECEIVE_BUFFER_SIZE);
+      bootstrap
+          .option(ChannelOption.SO_RCVBUF, receiveBufferSize)
+          .option(ChannelOption.RCVBUF_ALLOCATOR, new FixedRecvByteBufAllocator(receiveBufferSize));
+    }
+    if (config.isDefined(DefaultDriverOption.SOCKET_SEND_BUFFER_SIZE)) {
+      int sendBufferSize = config.getInt(DefaultDriverOption.SOCKET_SEND_BUFFER_SIZE);
+      bootstrap.option(ChannelOption.SO_SNDBUF, sendBufferSize);
+    }
   }
 
   @Override
```

Comment:
Not related to this PR, but I noticed that `PromiseCombiner` has been deprecated (used in method `onClose` below). Is there anything we can do to avoid using it?

Reply:
It's just the no-arg constructor that's deprecated. I guess we should inline its implementation: `new PromiseCombiner(ImmediateEventExecutor.INSTANCE)`; we can't use any of the driver's executors since they are being shut down.
I'll add the change in a separate commit.

Reply:
Actually `GlobalEventExecutor.INSTANCE` is a better choice (see the explanations on netty/netty@625c4e8286d3c8a36046feb1289b6b710c7541a5).